pkg/resourcemanager/vm/reconcile.go

```diff
@@ -91,10 +101,19 @@ func (g *AzureVirtualMachineClient) Ensure(ctx context.Context, obj runtime.Obje
 			switch azerr.Type {
 			case errhelp.AsyncOpIncompleteError:
 				instance.Status.Provisioning = true
+				instance.Status.Message = err.Error()
+
 			}
 			// reconciliation is not done but error is acceptable
 			return false, nil
 		}
+		//change to spec is required
+		if future.Response().StatusCode == 400 {
+			instance.Status.FailedProvisioning = true
+			instance.Status.Message = err.Error()
```

Comment:
same here, redundant setting of the error message

Reply:
removed